### ISSUE TITLE
TypeError: Tuple[t0, t1, ...]: each t must be a type. Got ().v

### ISSUE THREAD

**Issue by akayunov** [Instagram/MonkeyType#136]
Then I run "monkeytype apply" I got next error:

Traceback (most recent call last):
  File "/usr/local/bin/monkeytype", line 10, in <module>
    sys.exit(entry_point_main())
  File "/usr/local/lib/python3.7/site-packages/monkeytype/cli.py", line 381, in entry_point_main
    sys.exit(main(sys.argv[1:], sys.stdout, sys.stderr))
  File "/usr/local/lib/python3.7/site-packages/monkeytype/cli.py", line 366, in main
    handler(args, stdout, stderr)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/cli.py", line 140, in apply_stub_handler
    stub = get_stub(args, stdout, stderr)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/cli.py", line 127, in get_stub
    rewriter=rewriter,
  File "/usr/local/lib/python3.7/site-packages/monkeytype/stubs.py", line 597, in build_module_stubs_from_traces
    defn = get_updated_definition(func, traces, rewriter, existing_annotation_strategy)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/stubs.py", line 279, in get_updated_definition
    arg_types = {name: rewriter.rewrite(typ) for name, typ in arg_types.items()}
  File "/usr/local/lib/python3.7/site-packages/monkeytype/stubs.py", line 279, in <dictcomp>
    arg_types = {name: rewriter.rewrite(typ) for name, typ in arg_types.items()}
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 236, in rewrite
    typ = rw.rewrite(typ)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 148, in rewrite
    return rewriter(typ)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 130, in rewrite_Union
    return self._rewrite_container(Union, union)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 114, in _rewrite_container
    elems = tuple(self.rewrite(elem) for elem in container.__args__)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 114, in <genexpr>
    elems = tuple(self.rewrite(elem) for elem in container.__args__)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 148, in rewrite
    return rewriter(typ)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 127, in rewrite_Tuple
    return self._rewrite_container(Tuple, tup)
  File "/usr/local/lib/python3.7/site-packages/monkeytype/typing.py", line 115, in _rewrite_container
    return cls[elems]
  File "/usr/local/lib/python3.7/typing.py", line 743, in __getitem__
    return self.__getitem_inner__(params)
  File "/usr/local/lib/python3.7/typing.py", line 251, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.7/typing.py", line 769, in __getitem_inner__
    params = tuple(_type_check(p, msg) for p in params)
  File "/usr/local/lib/python3.7/typing.py", line 769, in <genexpr>
    params = tuple(_type_check(p, msg) for p in params)
  File "/usr/local/lib/python3.7/typing.py", line 139, in _type_check
    raise TypeError(f"{msg} Got {arg!r:.100}.")
TypeError: Tuple[t0, t1, ...]: each t must be a type. Got ().

=======================================
python3.7
MonkeyType         19.5.0

**Comment by carljm** [comment]
Seems like a bug in a type rewriter where it's trying to rewrite a tuple container to have zero elements. If you're able to isolate the code that triggers this, that would be super helpful; I've cross-checked the traceback against the code and it doesn't really clarify what kind of code (or traces) would cause us to end up processing a `Tuple` with no `__args__`.

**Comment by mykolav** [comment]
I think I'm seeing the same issue.
```
Traceback (most recent call last):  
  File "c:\users\username\miniconda3\envs\bs4\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\username\miniconda3\envs\bs4\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\username\Miniconda3\envs\bs4\Scripts\monkeytype.exe\__main__.py", line 9, in <module>
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\cli.py", line 381, in entry_point_main
    sys.exit(main(sys.argv[1:], sys.stdout, sys.stderr))
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\cli.py", line 366, in main
    handler(args, stdout, stderr)
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\cli.py", line 193, in print_stub_handler
    stub = get_stub(args, stdout, stderr)
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\cli.py", line 127, in get_stub
    rewriter=rewriter,
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\stubs.py", line 597, in build_module_stubs_from_traces
    defn = get_updated_definition(func, traces, rewriter, existing_annotation_strategy)
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\stubs.py", line 279, in get_updated_definition
    arg_types = {name: rewriter.rewrite(typ) for name, typ in arg_types.items()}
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\stubs.py", line 279, in <dictcomp>
    arg_types = {name: rewriter.rewrite(typ) for name, typ in arg_types.items()}
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\typing.py", line 236, in rewrite
    typ = rw.rewrite(typ)
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\typing.py", line 148, in rewrite
    return rewriter(typ)
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\typing.py", line 127, in rewrite_Tuple
    return self._rewrite_container(Tuple, tup)
  File "c:\users\username\miniconda3\envs\bs4\lib\site-packages\monkeytype\typing.py", line 115, in _rewrite_container
    return cls[elems]
  File "c:\users\username\miniconda3\envs\bs4\lib\typing.py", line 746, in __getitem__
    return self.__getitem_inner__(params)
  File "c:\users\username\miniconda3\envs\bs4\lib\typing.py", line 254, in inner
    return func(*args, **kwds)
  File "c:\users\username\miniconda3\envs\bs4\lib\typing.py", line 772, in __getitem_inner__
    params = tuple(_type_check(p, msg) for p in params)
  File "c:\users\username\miniconda3\envs\bs4\lib\typing.py", line 772, in <genexpr>
    params = tuple(_type_check(p, msg) for p in params)
  File "c:\users\username\miniconda3\envs\bs4\lib\typing.py", line 142, in _type_check
    raise TypeError(f"{msg} Got {arg!r:.100}.")
TypeError: Tuple[t0, t1, ...]: each t must be a type. Got ().

```

To reproduce follow these steps.
- download (beautifulsoup4-4.7.1.tar.gz)[https://www.crummy.com/software/BeautifulSoup/bs4/download/4.7/beautifulsoup4-4.7.1.tar.gz] and unpack it somewhere
- cd to the dir where you unpacked the source
- `conda create --name bs4 python`
- `source activate bs4`
- `conda install soupsieve lxml && pip install monkeytype`
- `./convert-py3k && cd py3k && monkeytype run -m unittest discover -s bs4`
- at this stage `monkeytype list-modules` should give you something along the lines
  > bs4
  > bs4.builder
  > bs4.builder._htmlparser
  > bs4.builder._lxml
  > bs4.dammit
  > bs4.element
  > bs4.testing
  > tests.test_builder_registry
  > tests.test_htmlparser
  > tests.test_lxml
  > tests.test_soup
  > tests.test_tree
- and finally invoking `monkeytype stub bs4.element` will produce the traceback posted at the top of comment.

Please let me know, if you need any additional info.

**Comment by KangOl** [comment]
It is due to a default argument set to an empty tuple.
Minimal reproducible code:

issue136.py
```python
#!/usr/bin/env python3
def foo(bar=()):
    pass
```

main.py
```python
#!/usr/bin/env python3
from issue136 import foo

foo()
```

```bash
$ monkeytype run main.py
$ monkeytype stub issue136
Traceback (most recent call last):
  ...
```

---
`bs4` has this pattern at https://bazaar.launchpad.net/~leonardr/beautifulsoup/bs4/view/head:/bs4/element.py#L1595

**Comment by carljm** [comment]
@KangOl Thanks for the simplest-case reproduction, that was very helpful! I've pushed a fix for this issue in #151.
